### PR TITLE
Add Apple Fn / Globe key

### PR DESCRIPTION
**Pull request by vinorodrigues**
## Description

-----

**&#9888; &nbsp; This is not a real PR, just a source placeholder for @fauxpark to test**
*(I will edit it and add documentation if admins so wish for actual submission, else it can be closed.)*

-----

Modern macOS no longer locks out the use of the Apple Fn (&#x1F310;) key, thus allowing users to leverage this functionality in trade for NKRO.

Entirely based on @fauxpark 's [Gist](https://gist.github.com/fauxpark/010dcf5d6377c3a71ac98ce37414c6c4) with 2 minor changes:
- `KC_APPLE_FN` is now keycode `0x00C3` *(since `0x00C1` is taken up by `KC_MISSION_CONTROL`)*
- `qmk generate-keycodes` did not generate the `IS_APPLE_FN` macro in `keycode.h`, I changed this for the auto-gen `IS_APPLE_FN_KEYCODE`.

No other attempt is made to change the code - meaning if one chooses to add the Apple Fn keys then one also needs to disable NKRO.

### In lieu of docs

To compile: 

Head on over to your keyboard keymap file and add the following two lines to the `rules.mk` file:
```make
NKRO_ENABLE = no
APPLE_FN_ENABLE = yes
```

Add the `KC_APFN` keycode to the appropriate spot on the `keymap.c` file.

### Fn/&#x1F310; key combos to test on macOS Monterey & macOS Ventura 

* `Fn`+`F` - toggle full screen mode
* `Fn`+`Up` / `Fn`+`Down` - Page scrolling
* `Fn`+`Right` / `Fn`+`Left` - End of, and Start of document scrolling
* `Fn`+`E` - Insert symbol or emoji
* `Fn`+`Q` - Quick note
* `Fn` + `Delete` - Forward delete *(same as `Contro`-`D`)*
* `Fn`+`F1`..`F12` - Media keys *(should also work on macOS Catalina)*
    * `F1` - Main screen brightness down *(for active screen use `F14`)*
    * `F2` - Main screen brightness up *(for active screen use `F15`)*
    * `F3` - Mission Control
    * `F4` - Launchpad
    * `F5` - Underglow Down *(obviously not on the QMK kb)*
    * `F6` - Underglow Up *(obviously not on the QMK kb)*
    * `F7` - Previous Track
    * `F8` - Play / Pause
    * `F9` - Next Track
    * `F10` - Mute
    * `F11` - Volume Down
    * `F12` - Volume Up

> Will depend on your System Preferences > Keyboard
> 
image

## Types of Changes

- [x] Core
- [ ] Bugfix
- [x] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation  ***[TODO]***

## Issues Fixed or Closed by This PR

* #378
* #2179
* #14781
* #15178
* #16651
* #19743

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.  ***[TODO]***
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: data/constants/keycodes/keycodes_0.0.2_basic.hjson [hunk 1]
```diff
@@ -15,6 +15,14 @@
             "aliases": [
                 "KC_LPAD"
             ]
+        },
+        "0x00C3": {
```
Comment (tzarc) [inline data/constants/keycodes/keycodes_0.0.2_basic.hjson:19]: Would need to be in a 0.0.3 file. 0.0.2 is already published and locked in.

File: quantum/action.c [hunk 2]
```diff
@@ -1188,6 +1210,11 @@ void debug_action(action_t action) {
         case ACT_USAGE:
             ac_dprintf("ACT_USAGE");
             break;
+#ifdef APPLE_FN_ENABLE
+        case ACT_APPLE_FN:
+            dprint("ACT_APPLE_FN");
```
Comment (tzarc) [inline quantum/action.c:1215]:
```suggestion
            ac_dprintf("ACT_APPLE_FN");
```

**Review by tzarc — CHANGES_REQUESTED** [review]
I'll leave the rest of the functionality to others, but there's some inconsistencies that'll need to be sorted out regardless.

File: data/constants/keycodes/keycodes_0.0.3_basic.hjson [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (drashna) [inline data/constants/keycodes/keycodes_0.0.3_basic.hjson:1]: needs a `keycodes_0.0.3.hjson` to properly generate new keycodes based on the basic hjson

**Comment by drashna** [comment]
I'm curious as to why this implementation over the other:
https://github.com/jhorology/keyboards/blob/main/patches/qmk_00_applefn.patch

Which seems to work with NKRO (which, IMO, would be preferable)

File: data/constants/keycodes/keycodes_0.0.3_basic.hjson (hunk 3 above)
Comment (vinorodrigues) [reply data/constants/keycodes/keycodes_0.0.3_basic.hjson:1]: ok.  i see what it does.   also generates a new macro, so leveraging that for `IS_APPLE_FN_KEYCODE`.

**Comment by vinorodrigues** [comment]
> why this implementation over the other

A couple of reasons:
1. I didn't know of that one.
2. I know fauxpark's work, but jhorology was not known to me
3. That patch requires `EXTRAKEY_ENABLE`
4. macOS has a few weirdnesses with NKRO - see #15178 *(which is maked "closed" but was not resolved as it turned into an unrelated support issue)*
5. macOS does not use NKRO anyway *(since it's own boards don't support it)*, and I made the assumption that anyone wanting this feature will compile a "to be used on macOS only" variation of the f/w

Having said that - I fully expect fauxpark to have his own take on this and I expect to close it anyway.

**Comment by vinorodrigues** [comment]
**After extensive testing** *[thanks @filterpaper, @drashna, @tzarc, @fauxpark]* **- it turns out that this still does not work without changing the PID/VID to one of Apple's keebs.**

This is however not the case if [Karabiner-Elements](https://karabiner-elements.pqrs.org/) is installed .. in which case any PID/VID __will work__.  As this dependence sits outside of QMK, and it's alternative *(to spoof an Apple keeb)* leaves makers legally exposed, it thus makes no sense to add this to core as it would not work for the general consumer.  Closing PR draft.

**Comment by mpicardvolvo** [comment]
Question about how to use this, I don't see it mentioned in the QMK docs anywhere, I am relatively beginner so I only now how to use the QMK Configurator online, and I don't see this key as an option.

**Pull request closed without merging**